i3pystatus/wifionice.py

```diff
@@ -0,0 +1,219 @@
+    url_on_click = ''
+    wifi_adapters = ['wlan0']
+    wifi_names = ['WiFi@DB', 'WIFIonICE']
+
+    settings = (
+        ("final_destination", "Information text for 'final destination has been reached'"),
+        ("format_offtrain", "Formatter for 'not on a train' (module hidden if `None` - no formatters available)"),
+        ("format_ontrain", "Formatter for 'on a train'"),
+        ("off_train_interval", "time between updates if no train is detected"),
+        ("on_train_interval", "time between updates while on a train"),
+        ("url_on_click", "URL to open when left-clicking the module"),
+        ("wifi_adapters", "List of wifi adapters the module should consider "
+                          "when detecting if you are in a train. Set to `None` "
+                          "to disable that functionality."),
+        ("wifi_names", "List of Wifi network names that should be considered 'on a train'."),
+    )
+
+    def _format_time(self, seconds):
+        if seconds is None:
+            return "?"
+        seconds = int(seconds)
+        components = []
+        if seconds >= 3600:
+            hours = int(seconds / 3600)
+            seconds -= hours * 3600
+            components.append('{}h'.format(hours))
+        if seconds >= 60:
+            minutes = int(seconds / 60)
+            seconds -= minutes * 60
+            components.append('{}m'.format(minutes))
+        if not components:
+            components.append('now')
+        return " ".join(components)
+
+    def _check_wifi(self):
+        if self.wifi_adapters is None:
+            self.logger.debug('self.wifi_adapters is None')
+            return True
+
+        from basiciw import iwinfo
+        for adapter in self.wifi_adapters:
+            self.logger.info(f'Checking {adapter} for compatible wifi network')
+            iwi = iwinfo(adapter)
+            for wifi in self.wifi_names:
+                if iwi['essid'].lower() == wifi.lower():
+                    self.logger.info(f'{adapter} uses {wifi} - success!')
+                    return True
+        self.logger.info('No matching wifi connection found')
+        return False
+
+    def _loop(self):
+        self.logger.debug('begin of _loop()')
+        while True:
+            self.logger.debug('new _loop()')
+            if self._check_wifi():
+                self.logger.info('On a train :)')
+                try:
+                    trip_info_req = urlopen('https://iceportal.de/api1/rs/tripInfo/trip')
+                    self.trip_info = loads(trip_info_req.read())['trip']
+
+                    ice_status_req = urlopen('https://iceportal.de/api1/rs/status')
+                    self.ice_status = loads(ice_status_req.read())
+                except Exception:
+                    self.trip_info = {}
+                    self.ice_status = {}
+
+                self.logger.debug(f'trip_info: {repr(self.trip_info)}')
+                self.logger.debug(f'ice_status: {repr(self.ice_status)}')
```

Comment:
Python string formatting has built-in support for invoking `repr`, using `!r`. So, these could both be done using:

```python
self.logger.debug(f'trip_info: {self.trip_info!r}')
self.logger.debug(f'ice_status: {self.ice_status!r}')
```